### PR TITLE
React to product-owners.yml changes

### PR DESCRIPTION
**Pull request by chadwhitacre**
To better support routing and triage of GitHub issues, we're now defining [product areas and their owners](https://www.notion.so/Parts-of-Sentry-and-Their-Owners-473791bae5bf43399d46093050b77bf0) in [a `product-owners.yml` file](https://github.com/getsentry/security-as-code/blob/main/rbac/lib/product-owners.yml) in the [`security-as-code`](https://github.com/getsentry/security-as-code/) repo. This PR sets up an automation to update the product areas in a few files in this repo when their definition changes over there. The workflows here are triggered by [a workflow in the `security-as-code` repo](https://github.com/getsentry/security-as-code/blob/main/.github/workflows/cascade-product-owners-yml-changes.yml).

Here's a parallel [PR in `sentry-docs`](https://github.com/getsentry/security-as-code/pull/85).

Here's [an example PR](https://github.com/getsentry/sentry/pull/48606) that this automation will create here in `sentry` (just the last commit on there, the rest are one-time testing garbage). Here's [a cleaner example in `sentry-docs`](https://github.com/getsentry/sentry-docs/pull/6866).

### CODE REVIEW

**Comment by chadwhitacre** [comment]
I generated https://github.com/getsentry/sentry/pull/48606 locally using the shell script.

**Comment by chadwhitacre** [comment]
> Missing: the "Product Area: What's New" label is missing from the repo. It will be created.
 > Added: the "Product Area: What’s new" label in the repo is not expected. It will be deleted.

This is fine, there are [no issues routed here yet](https://github.com/getsentry/sentry/issues?q=label%3A%22Product+Area%3A+What%E2%80%99s+new%22+).

**Comment by chadwhitacre** [comment]
Dropping this back to draft until `sentry-docs` is proven out.

**Comment by asottile-sentry** [comment]
can you test this with draft PRs please -- my inbox is entirely these this morning

**Comment by chadwhitacre** [comment]
Sure thing, sorry about that. Will see how much more testing is needed or if this is ready to go. :)

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 1]
```diff
@@ -41,9 +41,9 @@ body:
       label: Product Area
       description: Where in the product did you find the bug?
       options:
-        # Keep in sync with feature.yml and labels.yml
+        # begin product areas - autogenerated by bin/react-to-blah-blah-blah.py
```
Comment (hubertdeng123) [inline .github/ISSUE_TEMPLATE/bug.yml:44]: does `bin/react-to-blah-blah-blah.py` exist though? 👀

File: bin/react-to-product-owners-yml-changes.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+with open(".github/labels.yml", "w+") as fp:
+    fp.writelines(head)
+    fp.writelines(area_lines)
+    fp.writelines(tail)
+
+
+# Now for issue templates. Same game. Hang in there.
+
+BEGINNING = "        # begin product areas - autogenerated by bin/react-to-blah-blah-blah.py\n"
```
[the remark this answers is not in the document]
Comment (chadwhitacre) [reply bin/react-to-product-owners-yml-changes.py]: Done in 1b5d209eb28b8a8fdcda7997d684dfa417339db0. 👍

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 1 above)
Comment (chadwhitacre) [reply .github/ISSUE_TEMPLATE/bug.yml]: No it was a joke, sorry. Fixed in 1b5d209eb28b8a8fdcda7997d684dfa417339db0. 👍

**Review by hubertdeng123 — APPROVED** [review]

**Pull request merged**